amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnector.scala

```diff
@@ -23,17 +40,20 @@ private[amqp] trait AmqpConnectorLogic { this: GraphStageLogic =>
 
   final override def preStart(): Unit =
     try {
-      connection = settings.connectionProvider.get
-      channel = connection.createChannel()
-
       val connShutdownCallback = getAsyncCallback[ShutdownSignalException] { ex =>
         if (!ex.isInitiatedByApplication) failStage(ex)
       }
       val shutdownListener = new ShutdownListener {
         override def shutdownCompleted(cause: ShutdownSignalException): Unit = connShutdownCallback.invoke(cause)
       }
-      connection.addShutdownListener(shutdownListener)
-      channel.addShutdownListener(shutdownListener)
+
+      connection = {
+        val c = settings.connectionProvider.get
+        c.addShutdownListener(shutdownListener)
+        ConnectionWithShutdown(c, shutdownListener)
```

Comment:
I find weird adding the connection shutdown and then creating the `ConnectionWithShutdown` instead of doing everything inside the `ConnectionWithShutdown`